lib/blockchain.go

```diff
@@ -3787,6 +3807,38 @@ func (bc *Blockchain) AddInputsAndChangeToTransactionWithSubsidy(
 			spendAmount += txMeta.DeSoToSellNanos
 		}
 	}
+	// If this is a NFT Bid txn and the NFT entry is a Buy Now, we add inputs to cover the bid amount.
+	if txArg.TxnMeta.GetTxnType() == TxnTypeNFTBid && txArg.TxnMeta.(*NFTBidMetadata).SerialNumber > 0 {
+		txMeta := txArg.TxnMeta.(*NFTBidMetadata)
+		// Create a new UtxoView. If we have access to a mempool object, use it to
+		// get an augmented view that factors in pending transactions.
+		utxoView, err := NewUtxoView(bc.db, bc.params, bc.postgres)
+		if err != nil {
+			return 0, 0, 0, 0, errors.Wrapf(err,
+				"_computeInputsForTxn: Problem creating new utxo view: ")
+		}
+		if mempool != nil {
+			utxoView, err = mempool.GetAugmentedUniversalView()
+			if err != nil {
+				return 0, 0, 0, 0, errors.Wrapf(err,
+					"_computeInputsForTxn: Problem getting augmented UtxoView from mempool: ")
+			}
+		}
+
+		nftKey := MakeNFTKey(txMeta.NFTPostHash, txMeta.SerialNumber)
+		nftEntry := utxoView.GetNFTEntryForNFTKey(&nftKey)
+
+		if nftEntry != nil && nftEntry.IsBuyNow {
+			spendAmount += txMeta.BidAmountNanos
+			if txArg.ExtraData == nil {
+				txArg.ExtraData = make(map[string][]byte)
+			}
+			// We also set IsBuyNow to true in extra data
+			txArg.ExtraData[IsBuyNowKey] = []byte{BoolToByte(true)}
```

Comment:
I suggest that we get ride of this extra data on the NFT bid.  See my reasoning in the comment on ComputeTransactionMetadata() for NFTBids.